### PR TITLE
Add data model and methods for FlowRunInput

### LINKED ISSUE #11337: Add modeling for FlowRunInput to OSS
For human-in-the-loop we need a way to store key/value data, to that end we should add modeling for `FlowRunInput` that enables us to store key/value information associated with a flow run. We'll then use this data store to build out the interactivity piece.

### PR DESCRIPTION
**Pull request by bunchesofdonald**
This adds the `FlowRunInput` data model, schemas and CRUD methods to support the human-in-the-loop work.

Closes #11337 
### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed

### CODE REVIEW

File: src/prefect/server/database/orm_models.py [hunk 1]
```diff
@@ -1339,6 +1339,21 @@ class ORMVariable:
...
+class ORMFlowRunInput:
+    @declared_attr
+    def flow_run_id(cls):
+        return sa.Column(
+            UUID(),
+            nullable=False,
+        )
+
+    key = sa.Column(sa.String, nullable=False)
+    value = sa.Column(JSON, nullable=False)
```
Comment (chrisguidry) [inline src/prefect/server/database/orm_models.py:1352]: :fire: Good call making it JSON, that's maximally flexible

**Review by chrisguidry — COMMENTED** [review]
Super exciting!  My questions are mostly about the design, the implementation looks solid

File: src/prefect/server/database/migrations/versions/postgresql/2023_12_06_102745_8f83fc6d55eb_create_flow_run_input_table.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            prefect.server.utilities.database.JSON(astext_type=Text()),
+            nullable=False,
+        ),
+        sa.Column(
+            "flow_run_id", prefect.server.utilities.database.UUID(), nullable=False
+        ),
+        sa.PrimaryKeyConstraint("id", name=op.f("pk_flow_run_input")),
+        sa.UniqueConstraint(
+            "flow_run_id", "key", name=op.f("uq_flow_run_input__flow_run_id_key")
+        ),
```
[the remark this answers is not in the document]
Comment (bunchesofdonald) [reply src/prefect/server/database/migrations/versions/postgresql/2023_12_06_102745_8f83fc6d55eb_create_flow_run_input_table.py:54]: We discussed this synchronously, but the way this is setup is that the `flow_run_id` + `key` is unique per `FlowRunInput` so multiple suspends would have unique keys.

File: src/prefect/server/schemas/actions.py [hunk 3]
```diff
@@ -746,3 +746,25 @@ class VariableUpdate(ActionBaseModel):
...
+
+@copy_model_fields
+class FlowRunInputCreate(ActionBaseModel):
+    """Data used by the Prefect REST API to create a FlowRunInput"""
+
+    flow_run_id: UUID = FieldFrom(schemas.core.FlowRunInput)
+    key: str = FieldFrom(schemas.core.FlowRunInput)
+    value: str = Field(
+        description="The JSON value that should be stored for the input."
+    )
```
[the remark this answers is not in the document]
Comment (bunchesofdonald) [reply src/prefect/server/schemas/actions.py:759]: The server is agnostic about the serialization of the value that the user sending, it's just always treated as a string.

**Review by chrisguidry — APPROVED** [review]
Slow and steady wins the race!

**Comment by abrookins** [comment]
This looks good, and I think the simplified k/v store is better (from our DMs). Can you mark `test_increment_active_slots_with_decay_slots_decay_over_time` as a flaky test, since it failed based on what looks like timing issues?

**Review by abrookins — APPROVED** [review]

**Pull request merged**